src/quickhtml/style/css.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+from typing import Any, Generator, Union
+
+
+class StyleSheet:
+    def __init__(self, **styles):
+        self.__style_rules = styles
+
+    @property
+    def rules(self) -> dict:
+        return self.__style_rules
+
+    @classmethod
+    def from_css(cls, css: str | dict):
+        if isinstance(css, str):
+            k_v = {}
+            for statement in css.split(";"):
+                if not statement:
+                    continue
+                assert statement.count(":") == 1, f"Invalid CSS string '{css}'"
+
+                k, v = map(lambda x: x.strip(" :;\n"), css.split(":"))
```

Comment:
Should this be `statement` instead of `css`?
```suggestion
                k, v = map(lambda x: x.strip(" :;\n"), statement.split(":"))
```